wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceRuntimes.java

```diff
@@ -782,17 +795,22 @@ private boolean containsThisRuntimesId(String workspaceId) {
   }
 
   public Set<String> getSupportedRecipes() {
-    return environmentFactories.keySet();
+    return Sets.union(environmentFactories.keySet(), Collections.singleton("no-environment"));
```

Comment:
You are right. I forgot to do this

Reply:
@sleshchenko what's an example of infrastructure that doesn't support `no-environment`?

Reply:
For now, it's only docker infrastructure. I prefer to make approach flexible and allow not support running `no-environments` if it is possible.

Reply:
Ok I see. But that's only temporary. We are evaluating if Docker infrastructure will be supported or not in Che 7 but, if it will be supported, we have to implement `no-environment` for it as well. In other words we should not complicate the implementation to allow infrastructures that do not support `no-environment` (because all should support it).